src/mame/rm/rm380z_m.cpp

```diff
@@ -321,6 +325,16 @@ void rm380z_state::disk_0_control(uint8_t data)
 
 	if (floppy)
 	{
+		// FDS and MDS drives use a differnt square wave clock frequency
+		if (floppy->get_form_factor() == floppy_image::FF_8)
+		{
+			m_fdc->set_unscaled_clock(2_MHz_XTAL);
+		}
+		else
+		{
+			m_fdc->set_unscaled_clock(1_MHz_XTAL);
+		}
```

Comment:
Is the system actually capable of detecting the drive type and changing the floppy controller clock when selecting a drive like this?  Because if it isn’t, we shouldn’t be adding magical abilities like this in MAME.

Reply:
Fair comment, but Mame does already do this in at least one other driver.  Please see xerox/xerox820.cpp (line 267).  This is where I got the idea!  Does that create a precedent or would you still prefer it be fixed based upon machine type (2 Mhz for FDS, and 1 Mhz for MDS)?

With real hardware you needed a special machine with two FDC boards and modified firmware.  I believe these were available for organisations that needed to transfer data between MDS and FDS systems.  Although there is no dump of that firmware.

Reply:
There’s a lot of bad code in MAME.  That doesn’t mean we want more of it.  Just set the FDC clock in machine configuration.

Also, does it actually have a 1MHz or 2MHz crystal for the FDC, or does it get the FDC clock frequency by dividing some other clock?  You aren’t supposed to add crystals that aren’t actually present in the system, e.g. if it gets the FDC clock frequency by dividing the main 16MHz clock, you should use `16_MHz_XTAL / 16` or `16_MHz_XTAL / 8` to get the 1MHz or 2MHz frequency.

Reply:
Done!  There is no crystal so I've divided as you suggested.  I did wonder why clocks were divided like that throughout the code, but now I understand.